### PR TITLE
Vertex shaders & vertex loader -> x64 JIT

### PR DESCRIPTION
**Pull request by wheremyfoodat**

### CODE REVIEW

**Comment by wheremyfoodat** [comment]
image
OK maybe we're going in the wrong direction

**Comment by georgemoralis** [comment]
Hmm mostly dummy code atm but I am kinda excited!

**Comment by wheremyfoodat** [comment]
🥰

**Comment by Wunkolo** [comment]
Excited for this one! Lots of potential speed with stuff like AVX2/AVX512.

File: src/core/PICA/dynapica/shader_rec.cpp [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+void ShaderJIT::reset() {
+	cache.clear();
+}
+
+void ShaderJIT::prepare(PICAShader& shaderUnit) {
+	shaderUnit.pc = shaderUnit.entrypoint;
+	// We construct a shader hash from both the code and operand descriptor hashes
+	// This is so that if only one of them changes, we still properly recompile the shader
+	// This code is inspired from how Citra solves this problem
+	Hash hash = shaderUnit.getCodeHash() ^ shaderUnit.getOpdescHash();
```
Comment (Wunkolo) [inline src/core/PICA/dynapica/shader_rec.cpp:14]: This is a total nit-pick but xor is a commutative operation so behavior like `a^b == b^a` could produce the same hash here causing a possible collision-hazard. It's ridiculously low of course but you might want to do something like `std::rotl(x, 1) ^ y` to be more correct here.

Comment (wheremyfoodat) [reply src/core/PICA/dynapica/shader_rec.cpp:14]: Very good observation. I'll think about it for a sec and change it.

Comment (wheremyfoodat) [reply src/core/PICA/dynapica/shader_rec.cpp]: Added

**Comment by wheremyfoodat** [comment]
Superseded in #39

**Pull request closed without merging**